resources/assert_lib.brs

```diff
@@ -17,65 +17,90 @@ sub __equal(actual, expected, error)
     if actual = expected then
         m.__pass()
     else
-        m.__fail(m.formatError(error))
+        m.__fail(m.formatError({
+            message: error,
+            actual: actual,
+            expected: expected,
+            funcName: "m.assert.equal"
+        }))
     end if
 end sub
 
 sub __notEqual(actual, expected, error)
     if actual <> expected then
         m.__pass()
     else
-        m.__fail(m.formatError(error))
+        m.__fail(m.formatError({
+            message: error,
+            actual: actual,
+            expected: expected,
+            funcName: "m.assert.notEqual"
+        }))
     end if
 end sub
 
 sub __isTrue(actual, error)
     if actual = true then
         m.__pass()
     else
-        m.__fail(m.formatError(error))
+        m.__fail(m.formatError({
+            message: error,
+            actual: actual,
+            expected: true,
+            funcName: "m.assert.isTrue"
+        }))
     end if
 end sub
 
 sub __isFalse(actual, error)
     if actual = false then
         m.__pass()
     else
-        m.__fail(m.formatError(error))
+        m.__fail(m.formatError({
+            message: error,
+            actual: actual,
+            expected: false,
+            funcName: "m.assert.isFalse"
+        }))
     end if
 end sub
 
 sub __isInvalid(actual, error)
     if actual = invalid then
         m.__pass()
     else
-        m.__fail(m.formatError(error))
+        m.__fail(m.formatError({
+            message: error,
+            actual: actual,
+            expected: invalid,
+            funcName: "m.assert.isInvalid"
+        }))
     end if
 end sub
 
-function __formatError(errMessage)
-    return {
-        error: {
-            message: errMessage
-        },
-        stack: {}
-    }
+function __formatError(error)
+    ' Get the stack trace where the failed test is, filtering out any roca frames
+    fileFilters = ["roca"]
+    numStackFrames = 2
+    error.stack = _brs_.getStackTrace(numStackFrames, fileFilters)
+
+    return error
 end function
 
 sub __deepEquals(actual, expected, error)
     if __roca_deepEquals(actual, expected) = true then
         m.__pass()
     else
-        m.__fail(m.formatError(error))
+        m.__fail(m.formatError({
+            message: error
```

Comment:
Should we add `funcName: "m.assert.deepEquals` to this object?

Reply:
great catch!